src/m2n/tests/PointToPointCommunicationTest.cpp

```diff
@@ -305,6 +305,120 @@ BOOST_AUTO_TEST_CASE(MPIPortsCommunication,
   }
 }
 
+// we check whether connections between participants ranks are build correctly! 
+BOOST_AUTO_TEST_CASE(ConnectionTest, * testing::OnSize(4))
+{
+  
+  assertion(utils::Parallel::getCommunicatorSize() == 4);
+
+  // com::PtrCommunicationFactory cf(new com::SocketCommunicationFactory);
+  com::PtrCommunicationFactory cf(new com::MPIPortsCommunicationFactory);  
+  utils::MasterSlave::_communication = std::make_shared<com::MPIDirectCommunication>();
+  mesh::PtrMesh mesh(new mesh::Mesh("Mesh", 2, true));
+
+  switch (utils::Parallel::getProcessRank()) {
+  case 0: {
+    utils::Parallel::splitCommunicator("Fluid.Master");
+
+    utils::MasterSlave::_rank       = 0;
+    utils::MasterSlave::_size       = 2;
+    utils::MasterSlave::_masterMode = true;
+    utils::MasterSlave::_slaveMode  = false;
+
+    utils::MasterSlave::_communication->acceptConnection("Fluid.Master", "Fluid.Slave", utils::Parallel::getProcessRank());
+    utils::MasterSlave::_communication->setRankOffset(1);
+  
+    mesh->getConnectedRanks().push_back(0);
```

Comment:
Please also add a variant with different `connectedRanks`. Ideally, without duplicating too much code.

Reply:
done!

Reply:
With the current solution, we have lot's of duplicated code. Why no just looping through the different inputs?